platform/darwin/src/MGLStyle.mm

```diff
@@ -190,7 +190,11 @@ - (void)addLayer:(MGLStyleLayer *)layer
          layer];
     }
 
-    self.mapView.mbglMap->addLayer(std::unique_ptr<mbgl::style::Layer>(layer.layer));
+    try {
+        self.mapView.mbglMap->addLayer(std::unique_ptr<mbgl::style::Layer>(layer.layer));
```

Comment:
Added (tests) for `MGLStyle#insertLayer` and and `MGLMapView#insertCustomStyleLayerWithIdentifier`.